src/writer/untabber.rs

```diff
@@ -0,0 +1,49 @@
+use std::io::{Result, Write};
+use super::TAB;
+
+pub struct Untabber {
+    orig: Box<Write>,
+}
+
+impl Untabber {
+    pub fn new(orig: Box<Write>) -> Untabber {
+        Untabber{ orig: orig }
+    }
+}
+
+impl Write for Untabber {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        let v = buf.into();
+        match String::from_utf8(v) {
+            Ok(s) => {
+                let bytes = untab(&s).into_bytes();
+                let delta_len = bytes.len() - buf.len();
+                match self.orig.write(&bytes) {
+                    Ok(n) if n < delta_len => Ok(n),
+                    //hack for return normal len
+                    Ok(n) => Ok(n - delta_len),
```

Comment:
I agree that as we using only English code no these character appears.
Function modification by your example.
Only question, can I combine last 2 cases in last match?
```
    fn write(&mut self, buf: &[u8]) -> Result<usize> {
        let mut chunks = buf.split(|b| b == &b'\t').peekable();
        let mut len = 0;
        let tab_bytes = TAB.to_string().into_bytes();
        loop {
            match chunks.next() {
                Some(chunk) => match try!(self.orig.write(chunk)) {
                    0 if chunk.len() > 0 => return Ok(len),
                    n => len = len + n,
                },
                None => break,
            };
            if chunks.peek().is_some() {
                match try!(self.orig.write(&tab_bytes)) {
                    TAB_SIZE => len = len + 1,
                    0 => return Ok(len),
                    _ => unreachable!(),
                };
            } else {
                break;
            }
        }
        Ok(len)
    }
```